### DISCUSSION TITLE
How to transmit data on the duration of each web request to Prometheus?

### DISCUSSION THREAD

**Discussion by hortn** [javamelody/javamelody#1225] [Q&A]
Hello everyone. I really need some help.
Is it possible to transmit data on the duration of each web request to Prometheus?
In the Javamelody web interface, there is information about web requests, such as method, duration, and more. I would like to send the same information to Prometheus.
I have enabled data representation as JMX MBeans. In the jmx-exporter configuration, I specified the following:

lowercaseOutputLabelNames: true
lowercaseOutputName: true
whitelistObjectNames:
  - "net.bull.javamelody:type=CounterRequest,context=,name=http"

But I can't figure out how to write a pattern for the correct output of metrics in Prometheus format. I'm trying to understand how to write a pattern and if it's possible to transmit information about the duration of each individual web request.
Thanks.

**Comment by evernat** [comment]
Transmitting data on the duration of every web request in the statistics to Prometheus every minute may mean too much data. Perhaps you should not do that or else you may reduce the frequency.

If you still want to do that, you need to write some code using javamelody classes.
You may look at https://github.com/javamelody/javamelody/wiki/UserGuideAdvanced#jsp-tags-for-prometheus

and you can look at the following jsp tag which prints stats about a single request given its id: https://github.com/javamelody/javamelody/blob/master/javamelody-core/src/main/java/net/bull/javamelody/PrometheusCounterRequestTag.java
You could probably write your own servlet or a spring controller, it is simpler than a JSP tag.
